web-animations-2/Overview.bs

```diff
@@ -986,6 +1058,65 @@ condition:
 > *   the <a>animation effect</a> has a <a>parent group</a> and the
 >     <a>parent group</a> is <a>current</a>.
 
+Replace:
+
+> An <a>animation effect</a> is in the <dfn export for="animation effect">before
+> phase</dfn> if the animation effect's <a>local time</a> is not
+> <a>unresolved</a> and <em>either</em> of the following conditions are met:
+>
+> 1.  the <a>local time</a> is less than the <a>before-active boundary time</a>,
+>     <em>or</em>
+> 1.  the <a>animation direction</a> is "backwards" and the <a>local
+>     time</a> is equal to the <a>before-active boundary time</a>.
+
+with:
+
+> An <a>animation effect</a> is in the <dfn export for="animation effect">before
+> phase</dfn> if the animation effect's <a>local time</a> is not
+> <a>unresolved</a> and <em>any</em> of the following conditions are met:
+>
+> 1.  <a>animation</a>'s <a>hold phase</a> is [=timeline/before phase|before=],
+>     <em>or</em>
+> 1.  <a>animation</a>'s <a>hold phase</a> is unresolved, <em>and</em>,
+>     <a>animation</a>'s [=timeline phase=] is [=timeline/before phase|before=],
+>     <em>or</em>
+> 1.  <a>animation</a>'s <a>hold phase</a> is unresolved, <em>and</em>
```

Comment:
I think these conditions that the hold phase is unresolved may be causing a lot of issues where you set the hold phase to active which results in the animation phase being unable to resolve to before or after even if the local time is before / after the active duration.

Reply:
Yes, this last condition should apply provided hold is not after.

This will imply that the timeline's effective start and end offsets are hard limits on the active region even with negative delays. It'll still be possible for the timeline phase to be active and the effect phase to be before/after (with the fix for hold phase being active).

Reply:
Added a definition for "effective timeline phase" to simplify and correct the logic.